src/components/PostDetails.tsx

```diff
@@ -2,105 +1,122 @@
+import { FC, useEffect, useState } from 'react';
+
+import { Post } from '../types/Post';
+import { Comment } from '../types/Comment';
+
+import { deleteComment, getPostComments } from '../utils/api';
+
 import { Loader } from './Loader';
 import { NewCommentForm } from './NewCommentForm';
 
-export const PostDetails: React.FC = () => {
+interface Props {
+  post: Post;
+}
+
+export const PostDetails: FC<Props> = ({ post }) => {
+  const [comments, setComments] = useState<Comment[]>([]);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState(false);
+  const [isForm, setIsForm] = useState(false);
+
+  const handleDeleteComment = (commentId: number) => {
+    deleteComment(commentId).catch(() => {
+      setError(true);
+    });
+
+    setComments(currentComments =>
+      currentComments.filter(comment => comment.id !== commentId),
+    );
+  };
+
+  useEffect(() => {
+    if (!post.id) {
+      return;
+    }
+
+    setIsLoading(true);
+    setIsForm(false);
+    setError(false);
+    setComments([]);
+
+    getPostComments(post.id)
+      .then(setComments)
+      .catch(() => setError(true))
+      .finally(() => setIsLoading(false));
```

Comment:
Consider using a try-catch block for error handling when calling getPostComments. The checklist suggests using try-catch for error handling, which can provide more control over error management.